### ISSUE TITLE
Vue3 Tutorial Project does not build with pyodide-webpack-plugin

### ISSUE THREAD

**Issue by msebas** [pyodide/pyodide-webpack-plugin#6]
Hi,

I tried to use pyodide-webpack-plugin in a Vue3 Project. When I run `npm run serve` I get an error `ReferenceError: require is not defined in ES module scope, you can use import instead` from `node_modules/@pyodide/webpack-plugin/plugin.mjs`.

# Full Error Message

```
> vue-3-tutorial@0.1.0 serve
> vue-cli-service serve

 ERROR  ReferenceError: require is not defined in ES module scope, you can use import instead
ReferenceError: require is not defined in ES module scope, you can use import instead
    at Object.copy-webpack-plugin (file:///home/DABIT-ANALYTICS/sebastian.mueller/workspace/tools/tests/vue-3-tutorial/node_modules/@pyodide/webpack-plugin/plugin.mjs:84:1)
    at __webpack_require__ (file:///home/DABIT-ANALYTICS/sebastian.mueller/workspace/tools/tests/vue-3-tutorial/node_modules/@pyodide/webpack-plugin/plugin.mjs:128:40)
    at file:///home/DABIT-ANALYTICS/sebastian.mueller/workspace/tools/tests/vue-3-tutorial/node_modules/@pyodide/webpack-plugin/plugin.mjs:191:77
    at file:///home/DABIT-ANALYTICS/sebastian.mueller/workspace/tools/tests/vue-3-tutorial/node_modules/@pyodide/webpack-plugin/plugin.mjs:244:3
    at ModuleJob.run (node:internal/modules/esm/module_job:185:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:281:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:437:15)
    at async Service.run (/home/DABIT-ANALYTICS/sebastian.mueller/workspace/tools/tests/vue-3-tutorial/node_modules/@vue/cli-service/lib/Service.js:247:5)
```

# Reproduce

Scaffold a vue3 Project using
```
npm install -g @vue/cli@next
vue create vue-3-tutorial
```
(default vue3 configuration)
Add a `vue.confi.mjs` with 
```
import PyodidePlugin from "@pyodide/webpack-plugin";
import {defineConfig} from '@vue/cli-service';

export default defineConfig({
    configureWebpack: () => {return {plugins: [new PyodidePlugin()],}},
    transpileDependencies: true
})
```
and delete the `vue.config.js`.

Update `devDependencies` of the `package.json` to be 
```
{
    "@pyodide/webpack-plugin": "^1.0.0-alpha.1",
    "@babel/core": "^7.12.16",
    "@babel/eslint-parser": "^7.12.16",
    "@vue/cli-plugin-babel": "~5.0.0",
    "@vue/cli-plugin-eslint": "~5.0.0",
    "@vue/cli-service": "~5.0.0",
    "eslint": "^7.32.0",
    "eslint-plugin-vue": "^8.0.3",
    "pyodide": "^0.21.3"
  }
```
(e.g. add `@pyodide/webpack-plugin` and `pyodide`)

Run in the project directory
```
rm -rf node-modules
npm install
npm run serve
```

**Comment by mneil** [comment]
Thank you for the thorough report! I was able to recreate the issue with the .mjs module in this plugin and have pushed a fix in the last PR. I'm not able to publish this change yet but will do so tomorrow in a new release.

**Comment by msebas** [comment]
@mneil Thanks for the solution. It resolves the compile problem, but I stumbled upon a follow up loading problem, that might be related.

After building your fix `5d3ec27` I installed it to the test project. Trying to run the dev server using `npm run serve` I got an unrelated error stating that `Module not found: Error: Can't resolve 'url' in ...`, so I installed url using `npm install --save-dev url` and checked if `node_modules/@pyodide/webpack-plugin` was not updated by npm. Now the dev server starts up.
To do something with pyodide I updated `src/main.js` to be
```
import { createApp } from 'vue'
import App from './App.vue'
import {loadPyodide} from "pyodide";

createApp(App).mount('#app')
let pyodide = null;
async function main() {
  pyodide = await loadPyodide({ indexURL: `${window.location.origin}/pyodide` });
  return pyodide.runPython(`
    import sys
    sys.version
  `)
}
main().then(r => console.log(r));
```
(following the example in the `README.md`)

The dev-Server starts now, but loading `http://localhost:8080/` leads to a console error
```
Uncaught (in promise) Error: Cannot find module 'http://localhost:8080/pyodide/pyodide.asm.js'
    webpackEmptyAsyncContext app.js:99
    promise callback*webpackEmptyAsyncContext app.js:98
    loadScript compat.ts:158
    loadPyodide pyodide.ts:308
    main main.js:8
    <anonymous> main.js:15
    js app.js:63
    __webpack_require__ app.js:277
    __webpack_exports__ app.js:1499
    O app.js:326
    <anonymous> app.js:1500
    <anonymous> app.js:1502
app.js:99:11
```
which is cause by the function 
```
function webpackEmptyAsyncContext(req) {
	// Here Promise.resolve().then() is used instead of new Promise() to prevent
	// uncaught exception popping up in devtools
	return Promise.resolve().then(function() {
		var e = new Error("Cannot find module '" + req + "'");
		e.code = 'MODULE_NOT_FOUND';
		throw e;
	});
}
```
that seems to be loaded by pyodide using
`loadScript = async url => await __webpack_require__("./node_modules/pyodide lazy recursive")(url)`
where `url` is the string `"http://localhost:8080/pyodide/pyodide.asm.js"` (strangely calling the URL itself returns said script from the pyodide main module). 

Guessing that the value of `url` might be not valid for webpacks resolving mechanism I dropped the `indexURL` from `main.js` (loading pyodide via `pyodide = await loadPyodide();`). This resulted in the same error, only the value of `url` changed to `"webpack-internal:///./node_modules/pyodide/pyodide.asm.js"`.

I am a little bit at a loss what makes webpack screw up here, because the first value of `url` serves what is required.
